pygmt/src/rose.py

```diff
@@ -0,0 +1,212 @@
+"""
+rose - Plot windrose diagrams or polar histograms.
+"""
+
+from pygmt.clib import Session
+from pygmt.helpers import build_arg_string, fmt_docstring, kwargs_to_strings, use_alias
+
+
+@fmt_docstring
+@use_alias(
+    A="sector",
+    B="frame",
+    C="cmap",
+    D="shift",
+    Em="vectors",
+    F="no_scale",
+    G="color",
+    I="inquire",
+    JX="diameter",
+    L="labels",
+    M="vector_params",
+    Q="alpha",
+    R="region",
+    S="norm",
+    T="orientation",
+    U="timestamp",
+    V="verbose",
+    W="pen",
+    X="xshift",
+    Y="yshift",
+    Z="scale",
+    i="columns",
```

Comment:
The long alias for `i` might change from 'columns' to 'incols' or something (see https://github.com/GenericMappingTools/pygmt/issues/764 and links in there). Will need to discuss further.

Reply:
Ok, fine for me.

Reply:
For the sake of not keeping this PR open for too long, let's stick to using `i="columns"` now. But will need to remember to change it after there's a decision is made on #764 and before PyGMT v0.4.0 comes out.